File: Hello/hello2/src/main/java/com/example/hello2/MainActivity.java
package com.example.hello2;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Toast;
/*
    이 MainActivity 는 hello2 앱이 launch(실행)될때 최초로 사용자를 대면하는 앱티비티 이다.
    대체로 activity 하나는 화면 하나이다.
 */

public class MainActivity extends AppCompatActivity {
    // 액티비티가 활성화 될때 최초 호출되는 메소드
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // res/layout/activity_main.xml 문서를 전개(해석)해서 화면 구성하기
        setContentView(R.layout.activity_main);
    }

    // 버튼을 클릭했을때 호출되는 메소드 (view 객체의 참조값이 매개변수에 전달된다.
    public void clicked(View v){
        Toast.makeText(this, "버튼을 눌렀네요?", Toast.LENGTH_SHORT).show();
    }

    public void deleteClicked(View v){
        Toast.makeText(this, "삭제합니다.", Toast.LENGTH_SHORT).show();
    }
}

File: Hello/step08customview/src/main/java/com/example/step08customview/MainActivity.kt
package com.example.step08customview

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //setContentView(R.layout.activity_main)
        //MyView 객체를 생성해서 전달하기
        //setContentView(MyView(this))

        // layout xml 문서를 이용해서 MyView 객체를사용하려면
        // Context 와 AttributeSet 객체를 인자로 전달받는 생성자가
        // 존재 해야 한다.
        setContentView(R.layout.activity_main)

    }
}